src/network/networkd-sriov.c

```diff
@@ -103,3 +107,124 @@ int link_request_sr_iov_vfs(Link *link) {
 
         return 0;
 }
+
+static int link_find_sr_iov_phys_port(Link *link, Link **ret) {
+        _cleanup_(sd_device_unrefp) sd_device *pci_dev = NULL, *pci_physfn_dev = NULL;
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        const char *pci_syspath, *pci_physfn_syspath, *dev_port;
+        _cleanup_closedir_ DIR *dir = NULL;
+        sd_device **devices;
+        size_t n_devices;
+        bool found = false;
+        int r;
+
+        assert(link);
+        assert(link->manager);
+
+        if (link->phys_port_ifindex > 0)
+                return link_get_by_index(link->manager, link->phys_port_ifindex, ret);
+
+        if (link->phys_port_ifindex < 0) /* Previously parsed, and not found. */
+                return -ENODEV;
+
+        link->phys_port_ifindex = -1;
+
+        if (!link->sd_device)
+                return -ENODEV;
+
+        r = sd_device_get_sysattr_value(link->sd_device, "dev_port", &dev_port);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_parent_with_subsystem_devtype(link->sd_device, "pci", NULL, &pci_dev);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_syspath(pci_dev, &pci_syspath);
+        if (r < 0)
+                return r;
+
+        pci_physfn_syspath = strjoina(pci_syspath, "/physfn");
+        r = sd_device_new_from_syspath(&pci_physfn_dev, pci_physfn_syspath);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_syspath(pci_physfn_dev, &pci_physfn_syspath);
+        if (r < 0)
+                return r;
+
+        dir = opendir(pci_physfn_syspath);
+        if (!dir)
+                return -errno;
+
+        FOREACH_DIRENT_ALL(de, dir, break) {
+                _cleanup_free_ char *virtfn_link_file = NULL, *virtfn_pci_syspath = NULL;
+
+                if (isempty(startswith(de->d_name, "virtfn")))
```

Comment:
Is it intentional to check if the suffix is non-empty? `!startswith(…)` would be enough if not.